apps/stack/template.yaml

```diff
@@ -110,6 +116,22 @@ Parameters:
       Metrics Stream. If empty, no metrics will be collected.
     Default: 's3://observeinc/cloudwatchmetrics/filters/recommended.yaml'
     AllowedPattern: "^(s3:\/\/.*)?$"
+  ObserveAwsAccountId:
+    Type: String
+    Description: Observe AWS Account ID which will be allowed to assume role.
+    AllowedPattern: \d*
+  AllowedActions:
+    Type: CommaDelimitedList
+    Description: IAM actions that Observe account is allowed to execute.
+  DatastreamIds:
```

Comment:
is this check going to happen on the poller side? its not happening in the sam app side right?

Reply:
are you referring to the validity of the ObserveAwsAccountId? I thought this would happen based on the frontend passing in correct input. I'll hardcode AllowedActions

Reply:
This is because of how AWS recommends allowing other accounts to access your resources: see [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_third-party.html). It seems that it is possible to do this process without having an external id, but it is not recommended. This is also how the poller works currently - we pass in the datastreamIds ([terraform](https://github.com/observeinc/terraform-observe-aws-quickstart/blob/58a5f29435d9ec0d51b6490fc29d77931d5919ac/sources/cwmetricscraper/main.tf#L18)) and then we use it to assume roles. We take it as input [here](https://gerrit.observeinc.com/plugins/gitiles/observe/+/refs/heads/master/code/go/src/observe/ingest/poller/cloudwatchmetrics/poller.go#74) and use it to get credentials we can use in our polling. So the stack here is just checking that the request is coming from our backend.